cobra/core/solution.py

```diff
@@ -296,7 +296,11 @@ def get_solution(model, reactions=None, metabolites=None):
         forward = rxn.forward_variable.name
         reverse = rxn.reverse_variable.name
         fluxes[i] = var_primals[forward] - var_primals[reverse]
-        reduced[i] = var_duals[forward] - var_duals[reverse]
+        try:
+            reduced[i] = var_duals[forward] - var_duals[reverse]
+        except TypeError:
+            # reduced costs are not always defined, e.g. for integer problems
+            reduced[i] = nan
```

Comment:
Since the reduced cost is None in that case I think just testing with `if var_duals[forward]:` would be more explicit. Or test with `if var_duals[forward] and var_duals[reverse]:` but that is not necessary...